tests/k8st/tests/test_bgp_advert.py

```diff
@@ -165,7 +165,42 @@ def test_mainline(self):
         retry_until_success(lambda: self.assertIn(local_svc_ip, self.get_routes()))
         retry_until_success(lambda: self.assertNotIn(cluster_svc_ip, self.get_routes()))
 
-        # Delete nginx service
+        # Create a network policy that only accepts traffic from the external node
+        run("""kubectl exec -i -n kube-system calicoctl -- /calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-tcp-80-ex
+  namespace: bgp-test
+spec:
+  selector: all()
+  types:
+  - Ingress
+  ingress:
+  - action: Allow
+    protocol: TCP
+    source: { nets: ["10.192.0.5/32"] }
+    destination: { ports: ["80"] }
+EOF
+""")
+
+        # Connectivity to nginx-local should always succeed
+        self.do_external_curl("kube-node-extra", local_svc_ip)
+
+        # Connectivity to nginx-cluster will rarely succeed, try to curl 10 times
```

Comment:
Would be worth adding a "why" to this comment, for future reviewers.

e.g. "will rarely succeed because it is load-balanced across all nodes. When the traffic hits a node that doesn't host one of the service's pod, it will be re-routed to another node and SNAT will cause the policy to drop the traffic"